torch/_dynamo/variables/nn_module.py

```diff
@@ -366,7 +372,18 @@ def named_embed(name, obj):
                             **options,
                         )
                     )
-                return TupleVariable(result, **options)
+                    submods.append(submod)
+
+                new_module = torch.nn.Sequential(*submods)
+                new_module_variable = tx.output.register_attr_or_module(
+                    new_module,
+                    f"{self}.__getitem__(slice)",
+                    source=NNModuleSource(
+                        GetItemSource(self.source, args[0].as_python_constant())
+                    ),
+                    **options,
+                )
+                return NNSequentialVariable(new_module_variable, result, **options)
```

Comment:
@jansel I'm a little unsure about the keys in `result` and the new indices relative to new_module, but was not sure if I am thinking about this correctly. This passes tests, and we only use keys for naming, but this might look confusing as index keys mismatch their real positions. I added a test for this (reposition_indices in this PR). Still, not 100% sold this is doing the right thing...